libs/@hashintel/ds-components/src/components/Checkbox/checkbox.figma.tsx

```diff
@@ -0,0 +1,41 @@
+import figma from "@figma/code-connect";
+
+import { Checkbox } from "./checkbox";
+
+/**
+ * -- This file was auto-generated by Code Connect --
+ * `props` includes a mapping from your code props to Figma properties.
+ * You should check this is correct, and update the `example` function
+ * to return the code example you'd like to see in Figma
+ */
+
+figma.connect(
+  Checkbox,
+  "https://www.figma.com/design/WmnosvOvi4Blw0HK0jh1mG/Design-System?node-id=384%3A25638",
+  {
+    props: {
+      // These props were automatically mapped based on your linked code:
+      checked: figma.enum("checked", {
+        false: false,
+        true: true,
+        indeterminate: "indeterminate",
+      }),
+      disabled: figma.enum("_state", {
+        disabled: true,
+      }),
+      label: figma.string("label"),
+      // No matching props could be found for these Figma properties:
+      // "showLabel": figma.boolean('_showLabel'),
+      // "label": figma.string('label'),
+      // "showIcon": figma.boolean('_showIcon'),
+      // "icon": figma.instance('Icon')
```

Comment:
These should be discussed with @ciaranha.

I keep it in the Code Connect file, so we see what the API looks like on Figma, and we can possibly update it there.